authentication_document.py

```diff
@@ -215,6 +216,60 @@ def from_dict(cls, _db, data, place_class=Place):
             place_class=place_class
         )
 
+    def update_library(self, library):
+        """Modify a library to reflect the current state of this
+        AuthenticationDocument.
+        
+        :param library: A Library.
+        :return: A ProblemDetail if there's a problem, otherwise None.
+        """
+        problem = self.update_audiences(library)
+        if not problem:
+            problem = self.update_service_areas(library)
+        return problem
+
+        
+    def update_audiences(self, library):
+        return self._update_audiences(library, self.audiences)
+
+    @classmethod
+    def _update_audiences(self, library, audiences):
+        original_audiences = audiences
+        if not audiences:
+            audiences = [Audience.PUBLIC]
+        if isinstance(audiences, basestring):
+            # This is invalid but we can easily support it.
+            audiences = [audiences]
+        if not isinstance(audiences, list):
+            return INVALID_AUTH_DOCUMENT.detailed(
+                _("'audience' must be a list") % audiences
```

Comment:
String formatting doesn't work with flask-babel. You need to do `_("'audience' must be a list: %(audiences)r", audiences=audiences)`. Same for the other problem detail below.